src/lib/es2020.promise.d.ts

```diff
@@ -0,0 +1,30 @@
+interface PromiseResolution<T> {
+    status: "fulfilled";
+    value: T;
+}
+
+interface PromiseRejection {
+    status: "rejected";
+    reason: any;
+}
+
+type PromiseResult<T> = PromiseResolution<T> | PromiseRejection;

```

Comment:
I think this name is a bit too generic, in the sense that it could just as well be used for a `type PromiseResult<T extends PromiseLike<any>> = T extends PromiseLike<infer U> ? U : never` 🤔 (which, tbh, is useful enough to deserve to be a standard helper) 

Given how the spec language talks about "Resolve Element" or "Reject Element", perhaps qualifying all of these interfaces with an `Element` suffix would do. [`PromiseResolveElement`](https://tc39.es/ecma262/#sec-promise.allsettled-resolve-element-functions), [`PromiseRejectElement`](https://tc39.es/ecma262/#sec-promise.allsettled-reject-element-functions), and then `PromiseResultElement` for the union.

Reply:
I ultimately went with: `PromiseResolvedResult`, `PromiseRejectedResult` and `PromiseSettledResult`.